design/001-RPC-Interface-Specification.md

```diff
@@ -0,0 +1,413 @@
+<!-- This is a template for proposing design changes to the dst-go project. -->
+
+# Proposal: Descriptive specification of the User API for Payment Channels
+
+* Author(s): Manoranjith
+* Status: propose
+* Related issue:
+  [dst-go#45](https://github.com/direct-state-transfer/dst-go/issues/45)
+
+<!-- Use the above format for issues on GitHub and full links for issues on
+other platforms. -->
+
+## Summary
+
+<!-- Provide a tl;dr summary -->
+
+This specification describes the different APIs that should be provided by the
+node for enabling the user to open and use payment channels.
+
+## Motivation
+
+To have a standardized definition of the API interface, which can be used to
+derive protocol specific format specifications like Protocol Buffers (for gRPC)
+or JSON Schema for JSON RPC.
+
+## Details
+
+The node shall provide three sets of APIs:
+
+1. Node APIs - For accessing node related functionality of the node. Following
+APIs should be provided:
+
+    1. Get config
+    2. Open Session
+    3. Help
```

Comment:
Does it make sense to provide a list of available remote interface implementations / commands available and they will provide a self description if possible?
Like if they will run on the same port but different paths or maybe different ports...

Reply:
Yes, I think it makes sense to provide a list of APIs provided by a node instance. Currently I have updated the spec to include this information in the `Help` API.